sort/cyclesort.go

```diff
@@ -0,0 +1,59 @@
+package sort
+
+import (
+	"github.com/TheAlgorithms/Go/constraints"
+)
+
+// Cycle sort is an in-place, unstable sorting algorithm that is particularly useful
+// when sorting arrays containing elements with a small range of values. It is theoretically
+// optimal in terms of the total number of writes to the original array.
+func Cycle[T constraints.Number](arr []T) []T {
+
+	counter, cycle, len := 0, 0, len(arr)
```

Comment:
```suggestion
	counter, cycle, len := 0, 0, len(arr)
```

Reply:
This is a small nit so I'm committing - please make a pull on your fork to reflect these changes in your local 🙏🏼